factory/crypto/cryptoComponents.go

```diff
@@ -334,14 +329,16 @@ func (ccf *cryptoComponentsFactory) createCryptoParams(
 
 		return ccf.generateCryptoParams(keygen, "in import mode", handledPrivateKeys)
 	}
-	if ccf.noKeyProvided {
-		return ccf.generateCryptoParams(keygen, "with no-key flag enabled", make([][]byte, 0))
-	}
-	if len(handledPrivateKeys) > 0 {
-		return ccf.generateCryptoParams(keygen, "running with a provided allValidatorsKeys.pem", handledPrivateKeys)
+	cp, err := ccf.readCryptoParams(keygen)
+	if err == nil {
+		cp.handledPrivateKeys = handledPrivateKeys
+
+		return cp, nil
 	}
 
-	return ccf.readCryptoParams(keygen)
+	log.Debug("failure while reading the BLS key, will autogenerate one", "error", err)
+
+	return ccf.generateCryptoParams(keygen, "running without a provided BLS key", handledPrivateKeys)
```

Comment:
perhaps improve this message to mention if node is multikey or not? (and add one more similar warning on L335)

Reply:
done, the messages can be in this form (switched to INFO, btw)
```
the node loaded the validatorKey.pem file and is running in multi-key mode, managing 4 keys
the node loaded the validatorKey.pem file and is running in single-key mode 
the node is running in single-key mode! Will generate a fresh new BLS key 
the node is running in multi-key mode, managing 4 keys! Will generate a fresh new BLS key 
the node is in import-db mode! Will generate a fresh new BLS key 
```